### PR TITLE
Custom dateset support in big query

### PR DESCRIPTION
**Pull request by Jayachand**
## Description of the change

> Dojo customer wanted to custom dataset for bigquery driven through env customDatasetPrefix

## Notion Link

> https://www.notion.so/rudderstacks/Custom-dataset-in-namespace-specific-to-customer-dojo-7af8e82404364c41aedcb5aa13b61a29

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -105,6 +105,20 @@ const (
...
 )
 
+var whDestNameMap = map[string]string{
+	"BQ":             "bigquery",
+	"RS":             "redshift",
+	"MSSQL":          "mssql",
+	"POSTGRES":       "postgres",
+	"SNOWFLAKE":      "snowflake",
+	"CLICKHOUSE":     "clickhouse",
+	"DELTALAKE":      "deletalake",
```
Comment (achettyiitr) [inline warehouse/warehouse.go:115]: Typo deletalake

Comment (achettyiitr) [reply warehouse/warehouse.go:115]: Also, Can we move this to warehouse.utils and Keys can be replaced with Constants.

**Review by achettyiitr — APPROVED** [review]
LGTM

**Pull request merged**